gaphor/UML/actions/drop.py

```diff
@@ -0,0 +1,33 @@
+from gaphor.diagram.drop import drop
+from gaphor.diagram.presentation import connect
+from gaphor.diagram.support import get_diagram_item
+from gaphor.UML.uml import ActivityParameterNode
+
+
+@drop.register
+def drop_relationship(element: ActivityParameterNode, diagram, x, y):
+    item_class = get_diagram_item(type(element))
+    if not item_class:
+        return None
+
+    def drop_distance_to_item(item):
+        local_x, local_y = item.matrix_i2c.inverse().transform_point(x, y)
+        return item.point(local_x, local_y)
+
+    appendable_item = min(
+        diagram.select(
+            lambda item: element.activity and element.activity is item.subject
+        ),
+        key=drop_distance_to_item,
+        default=None,
+    )
+
+    if not appendable_item:
+        return None
+
+    item = diagram.create(item_class, parent=appendable_item, subject=element)
+    assert item
+    item.matrix.translate(x, y)
```

Comment:
I think it's better to make `(x, y)` relative to `appendable_item`.

Or change the code around a bit:

```python
item = diagram.create(item_class, subject=element)
item.matrix.translate(x, y)
item.change_parent(appendable_item)
```